src/dag_combine.rs

```diff
@@ -0,0 +1,56 @@
+use digraph::PyDiGraph;
+use pyo3::prelude::*;
+use petgraph::algo;
+use pyo3::Python;
+use pyo3::types::PyAny;
+use std::cmp::Ordering;
+
+fn contains_weight(py: Python, dag: &PyDiGraph, obj: &PyObject) -> i32 {
```

Comment:
I would probably change this to return a `PyResult<i32>`(or probably `PyResult<Option<usize>>` or `PyResult<Option<NodeIndex>>` see my comments below) so that you can return exceptions from the comparison back to python.